src/main/java/org/entando/kubernetes/security/AuthorizationFilter.java

```diff
@@ -0,0 +1,126 @@
+package org.entando.kubernetes.security;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.entando.kubernetes.model.web.response.SimpleRestResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class AuthorizationFilter extends HttpFilter {
+
+    public static final Pattern SERVLET_PATH_REGEX = Pattern.compile("^((/components)|(/registries)).*");
+    protected static final List<String> ECR_PERMISSION_LIST = Arrays.asList("superuser", "enterECR");
+    protected static final List<String> ALLOWED_METHODS = Arrays.asList(HttpMethod.PUT.name(), HttpMethod.POST.name(),
+            HttpMethod.DELETE.name());
+
+    private final RestTemplate restTemplate;
+    private final String entandoUrl;
+
+    public AuthorizationFilter(@Value("${entando.url}") final String entandoUrl, RestTemplate simpleRestTemplate) {
+        this.entandoUrl = entandoUrl;
+        this.restTemplate = simpleRestTemplate;
+    }
+
+    @Override
+    public void doFilter(
+            ServletRequest request,
+            ServletResponse response,
+            FilterChain chain) throws IOException, ServletException {
+
+        final HttpServletRequest httpRequest = (HttpServletRequest) request;
+        final String servletPath = httpRequest.getServletPath();
+        final String method = httpRequest.getMethod();
+
+        if (SERVLET_PATH_REGEX.matcher(servletPath).matches() && ALLOWED_METHODS.contains(method)) {
+
+            final String perm = fetchAndExtractRequiredPermission(httpRequest);
+
+            if (perm == null) {
+                HttpServletResponse resp = ((HttpServletResponse) response);
+                resp.sendError(HttpServletResponse.SC_FORBIDDEN, "The received JWT is not authorized");
```

Comment:
Maybe better `"Permissions included in the JWT don't allow the requested operation " + operation`